base-notebook/test/test_container_options.py

```diff
@@ -187,3 +187,30 @@ def test_group_add(container, tmpdir):
     assert rv == 0 or rv["StatusCode"] == 0
     logs = c.logs(stdout=True).decode("utf-8")
     assert "uid=1010 gid=1010 groups=1010,100(users)" in logs
+
+
+def test_container_not_delete_bind_mount(container, tmp_path):
+    """Container should not delete host system files when using the (docker)
+    -v bind mount flag and mapping to /home/jovyan.
+    """
+    d = tmp_path / "data"
+    d.mkdir()
+    p = d / "foo.txt"
+    p.write_text("some-content")
+
+    container.run(
+        tty=True,
+        user="root",
+        working_dir="/home/",
+        environment=[
+            "NB_USER=user",
+            "CHOWN_HOME=yes",
+        ],
+        volumes={d: {"bind": "/home/jovyan/data", "mode": "rw"}},
+        command=["start.sh", "jupyter", "lab"],
+    )
+
+    time.sleep(10)
```

Comment:
Let's wait with timeout here to not make this test always run 10 seconds?

It can be seen in the examples above I suppose.

Reply:
Hi! Yes, I did check that the test does fail with the original ```mv /home/jovyan "/home/${NB_USER}"``` command, and passes with the fix. As for the timeout I was following the above examples and following the pattern.

Reply:
I think the pattern above waits for 10 seconds in the worst case.
In general it works super-fast.
You're always waiting for 10 seconds.